umbra-js/src/classes/Umbra.ts

```diff
@@ -59,83 +62,59 @@ const isETH = (token: string) => {
   /**
    * @notice Create Umbra instance to interact with the Umbra contracts
-   * @dev WARNING: It is recommended that you do not call the constructor directly, and instead use
-   * the static methods to generate an Umbra instance. If you do call the constructor directly, be
-   * certain that the provided chainConfig information is valid
    * @param provider ethers provider to use
-   * @param signer The associated signer, or null if no signer is available
-   * @param chainConfig The chain configuration of the network
+   * @param chainConfig The chain configuration of the network or a network ID to use a default one
    */
-  constructor(
-    readonly provider: EthersProvider,
-    readonly signer: JsonRpcSigner | null,
-    readonly chainConfig: ChainConfig
-  ) {
-    const umbraAddress = chainConfig.umbraAddress;
-    this.umbraContract = new Contract(umbraAddress, abi, signer || provider) as UmbraContract;
+  constructor(readonly provider: EthersProvider, chainConfig: ChainConfig | number) {
+    this.chainConfig = parseChainConfig(chainConfig);
+    this.umbraContract = new Contract(
+      this.chainConfig.umbraAddress,
+      abi,
+      provider
+    ) as UmbraContract;
   }
 
-  /**
-   * @notice Generates Umbra instance from a web3 provider to read from and write to the Umbra contracts
-   * @param signer Signer to use
-   * @param chainConfig The chain configuration of the network, or null to read it from the signer
-   */
-  static async create(signer: JsonRpcSigner, chainConfig: ChainConfig | number | null = null) {
-    // If chainConfig is provided, use user-provided info
-    if (chainConfig) return new Umbra(signer.provider, signer, parseChainConfig(chainConfig));
-    // Otherwise, read chain ID from signer
-    const chainId = await signer.getChainId();
-    return new Umbra(signer.provider, signer, parseChainConfig(chainId));
-  }
+  // ==================================== CONTRACT INTERACTION =====================================
 
   /**
-   * @notice Generates Umbra instance from a JSON-RPC API to read from the Umbra contracts
-   * @param url URL of the JSON-RPC provider to connect to
-   * @param chainConfig The chain configuration of the network, or null to read it from the provider
+   * @notice Returns a signer with a valid provider
+   * @param signer Signer that may or may not have an associated provider
    */
-  static async createReadonly(url: string, chainConfig: ChainConfig | number | null = null) {
-    // If chainConfig is provided, use user-provided info
-    const provider = new JsonRpcProvider(url);
-    if (chainConfig) return new Umbra(provider, null, parseChainConfig(chainConfig));
-    // Otherwise, read chain ID from provider
-    const network = await provider.getNetwork();
-    return new Umbra(provider, null, parseChainConfig(network.chainId));
+  getSigner(signer: JsonRpcSigner | Wallet) {
```

Comment:
I see what this method is doing, but I wonder about the naming. Typically on an object a method starting with `get` would just be a param getter. I was confused by why I'd have to pass a signer into a method called `getSigner`. That said, I'm not sure what the right alternative is. Maybe `connectSigner` or `getConnectedSigner` or something like that?

Reply:
> Typically on an object a method starting with get would just be a param getter.

In ethers world this doesn't seem to be the case, as there are methods like `getTransactionReceipt()` and `getDefaultProvider()` which don't just return parameters.

I like `getConnectedSigner()` though, let's go with that

Reply:
Changed in ba0e6cbc548a145aa90dc54534b6baef8f192ff5